src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -615,7 +638,7 @@ public async Task<IActionResult> Reload(string id)
         [HttpPost]
         public async Task<IActionResult> Remove(string id)
         {
-            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageTenants))
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageTenants) || !_tenantsOptions.TenantRemovalAllowed)
```

Comment:
Display a notification instead when `!_tenantsOptions.TenantRemovalAllowed`. I'd argue the UI should be hidden too.

Reply:
In that case the "Remove" button is no more rendered, so here only for security.

Let me know if you still want the notification.